teamcity: failed tests on master: test/TestUpdateViewDependenciesMigration

The following tests appear to have failed:

[#314993](https://teamcity.cockroachdb.com/viewLog.html?buildId=314993):

```
--- FAIL: test/TestUpdateViewDependenciesMigration (10.240s)
migrations_test.go:535: starting server
test_server_shim.go:156: had 10 ranges at startup, expected 9
------- Stdout: -------
W170804 13:03:12.402870 404 server/server.go:299  [n?] all stores are configured as in-memory stores, so not setting up a temporary store. Queries with working set larger than memory will fail
W170804 13:03:12.403050 404 server/status/runtime.go:109  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I170804 13:03:12.408998 404 server/config.go:521  [n?] 1 storage engine initialized
I170804 13:03:12.409025 404 server/config.go:523  [n?] RocksDB cache size: 512 MiB
I170804 13:03:12.409034 404 server/config.go:523  [n?] store 0: in-memory, size 100 MiB
I170804 13:03:12.411635 404 server/node.go:335  [n?] **** cluster 82f5929f-db7b-456a-8685-46f971e22432 has been created
I170804 13:03:12.411665 404 server/server.go:784  [n?] **** add additional nodes by specifying --join=127.0.0.1:37104
I170804 13:03:12.412128 404 storage/store.go:1275  [n1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I170804 13:03:12.412171 404 server/node.go:421  [n1] initialized store [n1,s1]: diskAvailable=0 B/512 MiB (0.00%), ranges=1, leases=0, writes=0.00,bytesPerReplica={p10=3220.00 p25=3220.00 p50=3220.00 p75=3220.00 p90=3220.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00}
I170804 13:03:12.412200 404 server/node.go:319  [n1] node ID 1 initialized
I170804 13:03:12.412249 404 gossip/gossip.go:295  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:37104" > attrs:<> locality:<> 
I170804 13:03:12.412408 404 storage/stores.go:293  [n1] read 0 node addresses from persistent storage
I170804 13:03:12.412447 404 server/node.go:556  [n1] connecting to gossip network to verify cluster ID...
I170804 13:03:12.415659 404 server/node.go:581  [n1] node connected via gossip and verified as part of cluster "82f5929f-db7b-456a-8685-46f971e22432"
I170804 13:03:12.415688 404 server/node.go:362  [n1] node=1: started with [=] engine(s) and attributes []
I170804 13:03:12.415747 404 sql/executor.go:423  [n1] creating distSQLPlanner with address {tcp 127.0.0.1:37104}
I170804 13:03:12.442892 831 storage/replica_command.go:2667  [split,n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
I170804 13:03:12.444857 404 server/server.go:863  [n1] starting https server at 127.0.0.1:50212
I170804 13:03:12.444879 404 server/server.go:864  [n1] starting grpc/postgres server at 127.0.0.1:37104
I170804 13:03:12.444894 404 server/server.go:865  [n1] advertising CockroachDB node at 127.0.0.1:37104
W170804 13:03:12.444911 404 sql/jobs/registry.go:165  [n1] unable to get node liveness: node not in the liveness table
W170804 13:03:12.444921 404 sql/jobs/registry.go:290  [n1] canceling all jobs due to liveness failure
I170804 13:03:12.455879 831 storage/replica_command.go:2667  [split,n1,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/tsd [r3]
I170804 13:03:12.467104 404 server/server.go:1005  [n1] done ensuring all necessary migrations have run
I170804 13:03:12.467139 404 server/server.go:1007  [n1] serving sql connections
I170804 13:03:12.471223 665 sql/event_log.go:99  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:37104} Attrs: Locality:} ClusterID:82f5929f-db7b-456a-8685-46f971e22432 StartedAt:1501851792415674607 LastUp:1501851792415674607}
I170804 13:03:12.486829 831 storage/replica_command.go:2667  [split,n1,s1,r3/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r4]
I170804 13:03:12.502410 831 storage/replica_command.go:2667  [split,n1,s1,r4/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r5]
I170804 13:03:12.510843 831 storage/replica_command.go:2667  [split,n1,s1,r5/1:/{Table/System���-Max}] initiating a split of this range at key /Table/11 [r6]
I170804 13:03:12.523055 831 storage/replica_command.go:2667  [split,n1,s1,r6/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r7]
I170804 13:03:12.528559 831 storage/replica_command.go:2667  [split,n1,s1,r7/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r8]
I170804 13:03:12.543398 831 storage/replica_command.go:2667  [split,n1,s1,r8/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r9]
I170804 13:03:12.550575 831 storage/replica_command.go:2667  [split,n1,s1,r9/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r10]
I170804 13:03:22.619829 751 vendor/google.golang.org/grpc/grpclog/grpclog.go:75  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:37104->127.0.0.1:41211: use of closed network connection

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/migration/sqlmigrations/migrations_test.go
+++ b/pkg/migration/sqlmigrations/migrations_test.go
@@ -243,6 +243,9 @@ func TestEnsureMigrations(t *testing.T) {
 			0, 3, 0, 3,
 		},
 	}
+
+	defer func(prev []migrationDescriptor) { backwardCompatibleMigrations = prev }(backwardCompatibleMigrations)
+
 	for _, tc := range testCases {
 		t.Run("", func(t *testing.T) {
 			db.kvs = make(map[string][]byte)
@@ -312,6 +315,7 @@ func TestDBErrors(t *testing.T) {
 	defer mgr.stopper.Stop(context.TODO())
 
 	migration := noopMigration1
+	defer func(prev []migrationDescriptor) { backwardCompatibleMigrations = prev }(backwardCompatibleMigrations)
 	backwardCompatibleMigrations = []migrationDescriptor{migration}
 	testCases := []struct {
 		scanErr     error
@@ -375,6 +379,7 @@ func TestLeaseErrors(t *testing.T) {
 	defer mgr.stopper.Stop(context.TODO())
 
 	migration := noopMigration1
+	defer func(prev []migrationDescriptor) { backwardCompatibleMigrations = prev }(backwardCompatibleMigrations)
 	backwardCompatibleMigrations = []migrationDescriptor{migration}
 	if err := mgr.EnsureMigrations(context.Background()); err != nil {
 		t.Error(err)
@@ -421,6 +426,7 @@ func TestLeaseExpiration(t *testing.T) {
 			}
 		},
 	}
+	defer func(prev []migrationDescriptor) { backwardCompatibleMigrations = prev }(backwardCompatibleMigrations)
 	backwardCompatibleMigrations = []migrationDescriptor{waitForExitMigration}
 	if err := mgr.EnsureMigrations(context.Background()); err != nil {
 		t.Error(err)
@@ -441,12 +447,32 @@ func TestCreateSystemTable(t *testing.T) {
 	settingsDescVal := sqlbase.WrapDescriptor(&settingsDesc)
 
 	// Start up a test server without running the system.jobs migration.
-	backwardCompatibleMigrations = []migrationDescriptor{
-		{
-			name:   "create system.settings table",
-			workFn: createSettingsTable,
-		},
+	// remove the view update migration so we can test its effects.
+	newMigrations := append([]migrationDescriptor(nil), backwardCompatibleMigrations...)
+	for i := range newMigrations {
+		if strings.HasPrefix(newMigrations[i].name, "create system.jobs") {
+			// Disable.
+			//
+			// We merely replace the workFn and newDescriptors here instead
+			// of completely removing the migration step, because
+			// AdditionalInitialDescriptors needs to see the newRanges
+			// field. This is because another migration adds the web
+			// sessions table with larger ID 19. The split queue always
+			// splits at every table boundary, up to the maximum present
+			// table ID, even if the tables in-between are missing. So when
+			// table 19 is created, table ID 15 (the jobs table) will get a
+			// range, even though the table doesn't exist.
+			newMigrations[i].newDescriptors = 0
+			newMigrations[i].workFn = func(context.Context, runner) error { return nil }
+			// Change the name so that mgr.EnsureMigrations below finds
+			// something to do.
+			newMigrations[i].name = "disabled"
+			break
+		}
 	}
+	defer func(prev []migrationDescriptor) { backwardCompatibleMigrations = prev }(backwardCompatibleMigrations)
+	backwardCompatibleMigrations = newMigrations
+
 	s, _, kvDB := serverutils.StartServer(t, base.TestServerArgs{})
 	defer s.Stopper().Stop(ctx)
 
@@ -498,8 +524,10 @@ func TestCreateSystemTable(t *testing.T) {
 	// migration will get rerun here.
 	mgr := NewManager(s.Stopper(), kvDB, nil, s.Clock(), nil, "clientID")
 	backwardCompatibleMigrations = append(backwardCompatibleMigrations, migrationDescriptor{
-		name:   "create system.jobs table",
-		workFn: createJobsTable,
+		name:           "create system.jobs table",
+		workFn:         createJobsTable,
+		newDescriptors: 1,
+		newRanges:      1,
 	})
 	if err := mgr.EnsureMigrations(ctx); err != nil {
 		t.Fatal(err)
@@ -508,14 +536,14 @@ func TestCreateSystemTable(t *testing.T) {
 
 func TestUpdateViewDependenciesMigration(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	t.Skip("#17438")
-
 	ctx := context.Background()
 
 	// remove the view update migration so we can test its effects.
 	newMigrations := make([]migrationDescriptor, 0, len(backwardCompatibleMigrations))
 	for _, m := range backwardCompatibleMigrations {
 		if strings.HasPrefix(m.name, "establish conservative dependencies for views") {
+			// Remove this migration.
+			// This is safe because it neither adds descriptors nor ranges.
 			continue
 		}
 		newMigrations = append(newMigrations, m)
@@ -532,6 +560,8 @@ func TestUpdateViewDependenciesMigration(t *testing.T) {
 				return nil
 			},
 		})
+
+	defer func(prev []migrationDescriptor) { backwardCompatibleMigrations = prev }(backwardCompatibleMigrations)
 	backwardCompatibleMigrations = newMigrations
 
 	t.Log("starting server")

